src/mgmt/srv_drpc.c

```diff
@@ -1731,89 +1731,123 @@ pool_query_free_tier_stats(Mgmt__PoolQueryResp *resp)
 	resp->n_tier_stats = 0;
 }
 
+static int
+rank_list_to_str(d_rank_list_t *ranks, char **ranks_str)
+{
+	d_rank_range_list_t *range_list = NULL;
+	char                *range_list_str;
+	int                  rc;
+
+	D_ASSERT(ranks_str != NULL);
+
+	if (ranks == NULL) {
+		range_list_str = NULL;
+		D_GOTO(out, rc = -DER_SUCCESS);
+	}
+
+	range_list = d_rank_range_list_create_from_ranks(ranks);
+	if (range_list == NULL)
+		D_GOTO(error, rc = -DER_NOMEM);
+
+	rc = d_rank_range_list_str(range_list, &range_list_str);
+	if (rc != -DER_SUCCESS)
+		D_GOTO(error, rc = rc);
+
+out:
+	*ranks_str = range_list_str;
+
+error:
+	d_rank_range_list_free(range_list);
+
+	return rc;
+}
+
 void
 ds_mgmt_drpc_pool_query(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 {
-	struct drpc_alloc	alloc = PROTO_ALLOCATOR_INIT(alloc);
-	int			rc = 0;
-	Mgmt__PoolQueryReq	*req;
-	Mgmt__PoolQueryResp	resp = MGMT__POOL_QUERY_RESP__INIT;
-	Mgmt__StorageUsageStats	scm = MGMT__STORAGE_USAGE_STATS__INIT;
-	Mgmt__StorageUsageStats	nvme = MGMT__STORAGE_USAGE_STATS__INIT;
-	Mgmt__PoolRebuildStatus	rebuild = MGMT__POOL_REBUILD_STATUS__INIT;
-	uuid_t			uuid;
-	daos_pool_info_t	pool_info = {0};
-	d_rank_list_t		*svc_ranks;
-	d_rank_list_t		*ranks;
-	d_rank_range_list_t	*range_list;
-	char			*range_list_str = NULL;
-	bool			truncated;
-	size_t			len;
-	uint8_t			*body;
+	struct drpc_alloc       alloc = PROTO_ALLOCATOR_INIT(alloc);
+	int                     rc    = 0;
+	Mgmt__PoolQueryReq     *req;
+	Mgmt__PoolQueryResp     resp    = MGMT__POOL_QUERY_RESP__INIT;
+	Mgmt__StorageUsageStats scm     = MGMT__STORAGE_USAGE_STATS__INIT;
+	Mgmt__StorageUsageStats nvme    = MGMT__STORAGE_USAGE_STATS__INIT;
+	Mgmt__PoolRebuildStatus rebuild = MGMT__POOL_REBUILD_STATUS__INIT;
+	uuid_t                  uuid;
+	daos_pool_info_t        pool_info          = {0};
+	d_rank_list_t          *svc_ranks          = NULL;
+	d_rank_list_t          *enabled_ranks      = NULL;
+	d_rank_list_t          *disabled_ranks     = NULL;
+	char                   *enabled_ranks_str  = NULL;
+	char                   *disabled_ranks_str = NULL;
+	size_t                  len;
+	uint8_t                *body;
 
 	req = mgmt__pool_query_req__unpack(&alloc.alloc, drpc_req->body.len,
 					   drpc_req->body.data);
 	if (alloc.oom || req == NULL) {
-		D_ERROR("Failed to unpack pool query req\n");
+		D_ERROR("Failed to unpack pool query req");
```

Comment:
`D_ERROR` _does_ require '\n', I think; the `DL_` variants do not.

Reply:
- [ ] Add missing '\n' to D_ERROR function

Reply:
> * [x]  Add missing '\n' to D_ERROR function

Fixed with commit b8e199628